vivisect/analysis/generic/emucode.py

```diff
@@ -63,6 +64,9 @@ def iscode(self):
         return True
 
     def prehook(self, emu, op, eip):
+        if self.arch is None:
```

Comment:
first instruction we process (the first instruction of a function, if it succeeds) is the architecture we need to hand into `makeFunction()`

Reply:
the watcher would have to jump through all the hoops necessary to identify what architecture the thing is.  the only time we *know* that has to be the case is at first emulation.

Reply:
we could adjust the API?  or we could identify.